api/scpca_portal/test/test_loader.py

```diff
@@ -0,0 +1,311 @@
+        # different than the aggregate of the `diagnosis` values
+        # in the `sample_metadata.csv` with the test_data
+
+        # This is the value in the projects_metadata.csv file
+        # self.assertEqual(project.diagnoses, "Breast cancer")
+        # These are the values in the sample_metadata.csv file
+        self.assertIn("diagnosis1", project.diagnoses)
+        self.assertIn("diagnosis2", project.diagnoses)
+        self.assertIn("diagnosis5", project.diagnoses)
+        self.assertIn("diagnosis8", project.diagnoses)
+
+        # The following regex sequence pulls the diagnoses counts out of the project.diagnoses str,
+        # which is formatted as follows: "diagnosis1 (1), diagnosis2 (1), diagnosis5 (1), ..."
+
+        # Remove all words starting with 'diagnosis'
+        reg_ex_diagnoses_counts = re.sub(r"\bdiagnosis\d+\b", "", project.diagnoses_counts)
+        # Pull out all count totals left in regex string
+        diagnoses_counts_str_list = re.findall(r"\d+", reg_ex_diagnoses_counts)
+        # Cast them to ints and sum them up to arrive at a count
+        diagnoses_counts = sum([int(element) for element in diagnoses_counts_str_list])
+        self.assertEqual(diagnoses_counts, 4)
+
+        self.assertEqual(project.disease_timings, "Initial diagnosis")
+        # This value is not determined until after computed file generation
+        self.assertEqual(project.downloadable_sample_count, 0)
+        self.assertTrue(project.has_single_cell_data)
+        self.assertTrue(project.has_spatial_data)
+        self.assertEqual(project.human_readable_pi_name, "TBD")
+        self.assertTrue(project.includes_anndata)
+        self.assertFalse(project.includes_cell_lines)
+        self.assertTrue(project.includes_merged_sce)
+        self.assertTrue(project.includes_merged_anndata)
+        self.assertFalse(project.includes_xenografts)
+
+        self.assertIn(Sample.Modalities.NAME_MAPPING["BULK_RNA_SEQ"], project.modalities)
+        self.assertIn(Sample.Modalities.NAME_MAPPING["SPATIAL"], project.modalities)
+        self.assertEqual(project.multiplexed_sample_count, 0)
+        self.assertIn("Homo sapiens", project.organisms)
+        self.assertEqual(project.pi_name, "scpca")
+        self.assertEqual(project.s3_input_bucket, settings.AWS_S3_INPUT_BUCKET_NAME)
+
+        # single_cell samples: SCPCS999990, SCPCS999997
+        # spatial samples: SCPCS999991
+        # single_cell sample SCPCS999994 is unavailable
+        # As such, there should be 3 samples, but 4 are processed
+        # Like we do for projects where we check if a dir exists with data before processing,
+        # we should do the same for samples
+        # The check should be as follows
+        # single_cell, spatial = 2, 1
+        # self.assertEqual(project.sample_count, single_cell + spatial)
+
+        self.assertIn("cell", project.seq_units)
+        self.assertIn("spot", project.seq_units)
```

Comment:
Create a list, sort it, and then compare it to value in db which has been split and sorted